portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c

```diff
@@ -165,87 +165,150 @@ static u32 ulPHYIndex;
 
 /*-----------------------------------------------------------*/
 
+/* The function xNetworkInterfaceInitialise() will be called as
+ * long as it returns the value pdFAIL.
+ * It will go through several stages as described in 'eEMACState'.
+ */
+typedef enum xEMAC_STATE
+{
+    xEMAC_Init,
+    xEMAC_SetupPHY,
+    xEMAC_WaitPHY,
+    xEMAC_Ready,
+    xEMAC_Fatal,
+} EMACState_t;
+
+static EMACState_t eEMACState = xEMAC_Init;
+
 BaseType_t xNetworkInterfaceInitialise( void )
 {
     uint32_t ulLinkSpeed, ulDMAReg;
-    BaseType_t xStatus, xLinkStatus;
+    BaseType_t xStatus, xReturn = pdFAIL;
     XEmacPs * pxEMAC_PS;
-    const TickType_t xWaitLinkDelay = pdMS_TO_TICKS( 7000UL );
-    const TickType_t xWaitRelinkDelay = pdMS_TO_TICKS( 1000UL );
+    const TickType_t xWaitLinkDelay = pdMS_TO_TICKS( 1000U );
 
-    /* Guard against the init function being called more than once. */
-    if( xEMACTaskHandle == NULL )
+    switch( eEMACState )
     {
-        pxEMAC_PS = &( xEMACpsif.emacps );
-        memset( &xEMACpsif, '\0', sizeof( xEMACpsif ) );
+        case xEMAC_Init:
 
-        xStatus = XEmacPs_CfgInitialize( pxEMAC_PS, &mac_config, mac_config.BaseAddress );
+            ulPHYLinkStatus = 0U;
+            pxEMAC_PS = &( xEMACpsif.emacps );
```

Comment:
This pointer is only set in the case xEMAC_Init.
In the second xNetworkInterfaceInitialise() call it is used uninitialized in the xEMAC_SetupPHY case.

Reply:
Yes sure, good catch!
I just submitted this change.
Thanks

Reply:
lookin good now.